moveos/moveos-types/src/event_filter.rs

```diff
@@ -4,41 +4,39 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-use anyhow::Result;
-// use move_bytecode_utils::module_cache::GetModule;
-use move_core_types::account_address::AccountAddress;
-// use move_core_types::identifier::Identifier;
-// use move_core_types::language_storage::StructTag;
-use crate::event::Event;
+use crate::event::{Event, EventID};
 use crate::h256::H256;
 use crate::move_types::type_tag_match;
+use anyhow::Result;
+use move_core_types::account_address::AccountAddress;
 use move_core_types::language_storage::TypeTag;
-// use schemars::JsonSchema;
-use crate::object::ObjectID;
+use move_resource_viewer::AnnotatedMoveStruct;
 use serde::{Deserialize, Serialize};
 use serde_json::Value;
 use serde_with::serde_as;
 
 #[serde_as]
 // #[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize, JsonSchema)]
-#[derive(Eq, PartialEq, Clone, Debug)]
+// #[derive(Eq, PartialEq, Clone, Debug)]
+#[derive(Clone, Debug)]
 // #[serde(rename = "Event", rename_all = "camelCase")]
 pub struct MoveOSEvent {
-    pub key: ObjectID,
+    pub event_id: EventID,
     /// Sender's address.
     pub sender: AccountAddress,
     /// Transaction hash
-    pub tx_hash: H256,
+    pub tx_hash: Option<H256>,
     /// The number of messages that have been emitted to the path previously
-    pub sequence_number: u64,
+    // pub sequence_number: u64,
     // #[schemars(with = "String")]
     // #[serde_as(as = "TypeTag")]
     /// Move event type.
     pub type_tag: TypeTag,
     /// bcs bytes of the move event
     pub event_data: Vec<u8>,
     /// Parsed json value of the event data
-    pub parsed_json: Value,
+    // pub parsed_json: Value,
+    pub parsed_json: AnnotatedMoveStruct,
```

Comment:
I think we should define the `MoveOSEvent` and `AnnotatedMoveOSEvent`.  We can refactor it later.